gpytorch/models/exact_gp.py

```diff
@@ -45,6 +45,15 @@ def _apply(self, fn):
             self.train_targets = fn(self.train_targets)
         return super(ExactGP, self)._apply(fn)
 
+    def pyro_load_from_samples(self, samples_dict):
+        # Pyro always puts the samples in the first batch dimension
+        num_samples = next(iter(samples_dict.values())).size(0)
+        self.train_inputs = tuple(
+            tri.unsqueeze(0).repeat(num_samples, *([1] * len(tri.shape))) for tri in self.train_inputs
```

Comment:
do you want to repeat here or would expand do as well?

`tri.expand(num_samples, *tri.shape)`

Reply:
Yeah, expand should work